syntax = "proto3";
import "google/protobuf/timestamp.proto";

package object_state;

service AirflowToStateManager {
  rpc BatchImport (stream RequestBatchImport) returns (ResponseBatchImport) {}
}

message KPI{
  int64 kpi_id = 1;
  int64 object_id = 2;
  int64 granularity_id = 3;
  string value = 4;
  google.protobuf.Timestamp record_time = 5;
  States state = 6;
}

enum States {
  current = 0;
  historical = 1;
  planned = 2;
}

message RequestBatchImport {
    repeated KPI kpi_data = 1;
}



message ResponseBatchImport {
    string status = 1;
    optional string message = 2;
}
